transports/astartehttpendpoint.cpp

```diff
@@ -164,6 +164,10 @@ void HTTPEndpointPrivate::ensureCredentialsSecret()
     provider->setSslConfiguration(sslConfiguration);
     credentialsSecretProvider = provider;
 
+    if (!this->credentialsSecret.isEmpty()) {
+        provider->saveCredentialsSecret(this->credentialsSecret);
+    }
+
```

Comment:
Move this above
```
credentialsSecretProvider = provider
```
for consistency (we set everything up using `provider` and then we assign it)